### PR TITLE
Create shuffle algorithm

### PR DESCRIPTION
**Pull request by seisuke**
Add testClass

Move application.properties to application.yml

### CODE REVIEW

**Comment by Kiramind** [comment]
Good !
Could the max group member be parametrised ?

File: src/main/java/com/shufflelunch/service/GroupService.java [hunk 1]
```diff
@@ -46,4 +50,61 @@ public boolean deleteGroup(String name) {
...
+        int minIndex = memberNum.indexOf(Collections.min(memberNum));
+        memberNum.set(maxIndex, memberNum.get(maxIndex) - 1);
+        memberNum.set(minIndex, memberNum.get(minIndex) + 1);
+        return memberNum;
+    }
+
+    boolean validGrouping(List<Integer> memberList) {
+        int max = Collections.max(memberList);
+        int min = Collections.min(memberList);
+        return max - min < 2;
```
Comment (adamu) [inline src/main/java/com/shufflelunch/service/GroupService.java:91]: I think `return min > 2` is the same, and simpler!

**Comment by adamu** [comment]
LGTM!

**Comment by seisuke** [comment]
Now we can not parametarize it. 😢

**Comment by adamu** [comment]
```java
public List<Integer> calcMemberNumber(int memberSize, int defaultSize)
```

...
```java
    calcMemberNumber(n, 4)?
```

**Pull request merged**